extensions/amp-analytics/amp-analytics.md

```diff
@@ -87,6 +89,37 @@ At the top level, the config has following properties:
     -  **vars**: This property is same as the property of same name in parent block. Values specified here take precedence over the values specified in the platform and the parent block. 
 
 
+##### <a name="vars-substitution"></a>Variable Substitution and Formatting
+Variables can be specified in multiple places in the analytics config. Some of the variables are predefined by the platform. In addition, variables can also be defined, overridden or cleared by the config. Values set at the top level get applied to all the triggers and the values specified inside a trigger only apply to that trigger. The values of variables get encoded before the substitution using `encodeURIComponent`. The precedence for variable values is in the order trigger.vars > config.vars > platform.vars.
```

Comment:
Are you talking about the fact that amp-analytics delegates to URL
replacement for most platform variables or that amp analytics also replaces
variables in the format described in that var-substitutions?

The first is an implementation detail and the list of variables with camel
casing should be in this doc.

The second leads to confusion. There is very little reason for some vars to
be caps and others to be camel cased. It would be nice to just have one or
the other.

I see that you guys are discussing this in Brad's PR. I can submit this and
Brad can make things consistent or I can wait till you guys reach a
consensus.

Avi
On Dec 12, 2015 2:58 AM, "Dima Voytenko" <notifications@github.com> wrote:

> In extensions/amp-analytics/amp-analytics.md
> <https://github.com/ampproject/amphtml/pull/1140#discussion_r47407356>:
>
> > @@ -87,6 +89,37 @@ At the top level, the config has following properties:
> >      -  **vars**: This property is same as the property of same name in parent block. Values specified here take precedence over the values specified in the platform and the parent block.
> >
> >
> > +##### <a name="vars-substitution"></a>Variable Substitution and Formatting
> > +Variables can be specified in multiple places in the analytics config. Some of the variables are predefined by the platform. In addition, variables can also be defined, overridden or cleared by the config. Values set at the top level get applied to all the triggers and the values specified inside a trigger only apply to that trigger. The values of variables get encoded before the substitution using `encodeURIComponent`. The precedence for variable values is in the order trigger.vars > config.vars > platform.vars.
>
> Please explain how this interoperates with
> https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md
> and link it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ampproject/amphtml/pull/1140/files#r47407356>.
>